src/NServiceBus.Extensions.Hosting.Tests/NServiceBusHostedServiceTests.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus.Extensions.Hosting.Tests
+{
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class NServiceBusHostedServiceTests
+    {
+        [Test]
+        public void When_stopping_without_starting_should_not_throw()
```

Comment:
The unit test feels bit unnecessary tbh but given it's easy to keep a (different) NRE from happening by just switching to `endpoint?.Stop(cancellationToken)` it might be worth the few extra lines 🤷